### ISSUE TITLE
Initializing a workflow does not reindex the object

### ISSUE THREAD

**Issue by LynnMcRae** [sul-dlss/argo#72]
Adding a new workflow to the object either via Blue Button or direct service /gem call ...

To reproduce, use dor-services gem to do this: 

  i = Dor::Item.find('druid:aa111bb2222') 
  i.initialize_workflow('abcWF') 

Using a legit workflow. Goto Argo for that druid and notice that the workflow abcWF is not present in the History > Workflows section on the page. 

Click Reindex, and notice that the History > Workflows section is now correct.  Adding a new workflow should send an object update message to the DOR index queue.

**Comment by drh-stanford** [comment]
I've noticed this problem in production Argo as well. When I create fire off new workflows, I have to reindex to see the change in the workflows list. And reindexing fixes this. I had assumed it was a lag in the messaging. I'll take a look at the code path for this...

**Comment by drh-stanford** [comment]
I'm having difficulty tracking down this issue. I think I've narrowed down the location of the error (a missing reindex call in `ItemsController#add_workflow`), but I haven't been able to test my proposed fix.

which is to add `add_workflow` to the after trigger for save and reindex...
```
-  after_filter  :save_and_reindex,   :only => [:add_collection, :set_collection, :remove_collection, :open_version, :close_version, :tags, :tags_bulk, :source_id, :datastream_update, :set_rights, :set_content_type, :apply_apo_defaults]
+  after_filter  :save_and_reindex,   :only => [:add_collection, :set_collection, :remove_collection, :open_version, :close_version, :tags, :tags_bulk, :source_id, :datastream_update, :set_rights, :set_content_type, :apply_apo_defaults, :add_workflow]
```

**Comment by drh-stanford** [comment]
The problem is that the `RSolr.add` is not committing before the `/catalog` view page is coming up. I have a fix for this that is different than above. Will do a PR for it...

**Comment by drh-stanford** [comment]
there is a PR pending for this fix #145
